packages/node_modules/@webex/plugin-meetings/src/locus-info/parser.js

```diff
@@ -1,118 +1,708 @@
+
+    return hasEmptyEntries && hasEmptyRange;
+  }
+
+
+  /**
+   * Determines if an object has basic
+   * structure of a locus object.
+   * @param {Types~Locus} loci
+   * @returns {boolean}
+   */
+  static isLoci(loci) {
+    if (!loci || !loci.sequence) {
+      return false;
+    }
+    const hasProp = (prop) =>
+      Object.prototype.hasOwnProperty.call(loci.sequence, prop);
+
+    if (hasProp('rangeStart') && hasProp('rangeEnd')) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Processes next event in queue,
+   * if queue is empty sets status to idle.
+   * @returns {undefined}
+   */
+  nextEvent() {
+    if (this.status === Parser.status.PAUSED) {
+      LoggerProxy.logger.info('Locus-info:parser#nextEvent --> Locus parser paused.');
+
+      return;
+    }
+
+    // continue processing until queue is empty
+    if (this.queue.size() > 0) {
+      this.processDeltaEvent();
+    }
+    else {
+      this.status = Parser.status.IDLE;
+    }
+  }
+
+
+  /**
+   * Function handler for delta actions,
+   * is set by instance callee.
+   * @param {string} action Locus delta action
+   * @param {Types~Locus} locus Locus delta
+   * @returns {undefined}
+   */
+  // eslint-disable-next-line no-unused-vars
+  onDeltaAction(action, locus) {}
+
+
+  /**
+   * Event handler for locus delta events
+   * @param {Types~Locus} loci Locus Delta
+   * @returns {undefined}
+   */
+  onDeltaEvent(loci) {
+    // enqueue the new loci
+    this.queue.enqueue(loci);
+    // start processing events in the queue if idle
+    // and a function handler is defined
+    if (this.status === Parser.status.IDLE && this.onDeltaAction) {
+      // Update status, ensure we only process one event at a time.
+      this.status = Parser.status.WORKING;
+
+      this.processDeltaEvent();
+    }
+  }
+
+
+  /**
+   * Appends new data onto a string of existing data.
+   * @param {string} newData
+   * @param {string} oldData
+   * @returns {string}
+   */
+  static packComparisonResult(newData, oldData) {
+    return `${newData}:${oldData}`;
+  }
+
+
+  /**
+   * Pause locus processing
+   * @returns {undefined}
+   */
+  pause() {
+    this.status = Parser.status.PAUSED;
+    LoggerProxy.logger.info('Locus-info:parser#pause --> Locus parser paused.');
+  }
+
+
+  /**
+   * Processes next locus delta in the queue,
+   * continues until the queue is empty
+   * or cleared.
+   * @returns {undefined}
+   */
+  processDeltaEvent() {
+    const {DESYNC, USE_INCOMING} = Parser.loci;
+    const {extractComparisonState: extract} = Parser;
+    const newLoci = this.queue.dequeue();
+
+    if (!this.isValidLocus(newLoci)) {
+      return;
+    }
+
+    const result = Parser.compare(this.workingCopy, newLoci);
+    const lociComparison = extract(result);
+
+    if (this.debug) {
+      // limited debugging, use chrome extension
+      // for full debugging.
+      LoggerProxy.logger.info(`Locus-info:parser#processDeltaEvent --> Locus Debug: ${result}`);
```

Comment:
Could we just use log levels to determine if the use wants debug info and change this to:

```suggestion
      LoggerProxy.logger.debug(`Locus-info:parser#processDeltaEvent --> Locus Debug: ${result}`);
```

Reply:
Sure np, I removed .info and used .debug and also removed the conditional.